app/assets/javascripts/richtext.js

```diff
@@ -34,15 +34,54 @@ $(document).ready(function () {
     event.preventDefault();
   });
 
+  /*
+   * Install a click handler to return 'required' attribute for
+   * message_title (Subject) when send button is pressed.
+   */
+  $(".btn-primary[id=\"send_message_button_id\"]").click(function () {
+    var messageTitleElement = document.getElementById("message_title");
+    if (messageTitleElement) {
+      messageTitleElement.setAttribute("required", "required");
+    }
+  });
+
+  /*
+   * Install a click handler to return 'required' attribute to
+   * diary_entry_title (Subject) element, .. when publish / update
+   * button is pressed.
+   */
+  $(".btn-primary[id=\"publish_or_update_diary_entry_button_id\"]").click(function () {
+    var diaryEntryTitleElement = document.getElementById("diary_entry_title");
+    if (diaryEntryTitleElement) {
+      diaryEntryTitleElement.setAttribute("required", "required");
+    }
+  });
+
   /*
    * Install a click handler to switch to preview mode when the
-   * preview button is pressed.
+   * preview button is pressed. If editor is empty, remove 'required'
+   * attribute from message_title and diary_entry_title element and
+   * do early return (to trigger validation and error reporting)
    */
   $(".richtext_dopreview").click(function (event) {
     var editor = $(this).parents(".richtext_container").find("textarea");
     var preview = $(this).parents(".richtext_container").find(".richtext_preview");
     var minHeight = editor.outerHeight() - preview.outerHeight() + preview.height();
 
+    if (editor.val().length === 0) {
+      var messageTitleElement = document.getElementById("message_title");
+      if (messageTitleElement) {
+        messageTitleElement.removeAttribute("required");
+      }
+
+      var diaryEntryTitleElement = document.getElementById("diary_entry_title");
+      if (diaryEntryTitleElement) {
+        diaryEntryTitleElement.removeAttribute("required");
+      }
+
+      return;
+    }
+
```

Comment:
I don't necessarily agree that triggering the validation is a good idea, but if you really want to do it, you can just call `reportValidity` on an input:

```suggestion
    event.preventDefault();

    if (!editor[0].reportValidity()) {
      return;
    }
```